xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-api/src/main/java/org/xwiki/index/migration/R140300001XWIKI19571DataMigration.java

```diff
@@ -0,0 +1,184 @@
+import org.hibernate.engine.jdbc.connections.spi.JdbcConnectionAccess;
+import org.hibernate.engine.spi.SessionImplementor;
+import org.hibernate.mapping.PersistentClass;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.doc.tasks.XWikiDocumentIndexingTask;
+import org.xwiki.index.internal.TasksStore;
+
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.internal.store.hibernate.HibernateStore;
+import com.xpn.xwiki.store.migration.DataMigrationException;
+import com.xpn.xwiki.store.migration.XWikiDBVersion;
+import com.xpn.xwiki.store.migration.hibernate.AbstractHibernateDataMigration;
+
+/**
+ * Migrate XWikiDocumentIndexingTask by copying the table to a new temporary table. Then, after hibernate migration,
+ * when XWikiDocumentIndexingTask is created again, the rows from the temporary table are copied to the new table.
+ * Finally, the temporary table is dropped.
+ *
+ * @version $Id$
+ * @since 14.3RC1
+ */
+@Component
+@Singleton
+@Named(R140300001XWIKI19571DataMigration.HINT)
+public class R140300001XWIKI19571DataMigration extends AbstractHibernateDataMigration
+{
+    /**
+     * The hint for this component.
+     */
+    public static final String HINT = "R140300001XWIKI19571";
+
+    @Inject
+    private HibernateStore hibernateStore;
+
+    @Inject
+    private Provider<TasksStore> tasksStore;
+
+    private List<XWikiDocumentIndexingTask> tasks = new ArrayList<>();
+
+    @Override
+    public String getDescription()
+    {
+        return "Save the content of xwikidocumentindexingqueue, remove it, let Hibernate recreate it with a new schema"
+            + " and finally re-insert the saved content in xwikidocumentindexingqueue";
+    }
+
+    @Override
+    public XWikiDBVersion getVersion()
+    {
+        return new XWikiDBVersion(140300001);
+    }
+
+    @Override
+    public String getPreHibernateLiquibaseChangeLog() throws DataMigrationException
+    {
+        SessionFactory sessionFactory = this.hibernateStore.getSessionFactory();
+        try (SessionImplementor session = (SessionImplementor) sessionFactory.openSession()) {
+            JdbcConnectionAccess jdbcConnectionAccess = session.getJdbcConnectionAccess();
+            try (Connection connection = jdbcConnectionAccess.obtainConnection()) {
+                DatabaseMetaData databaseMetaData = connection.getMetaData();
+
+                PersistentClass persistentClass =
+                    this.hibernateStore.getConfigurationMetadata()
+                        .getEntityBinding(XWikiDocumentIndexingTask.class.getName());
+                String tableName = this.hibernateStore.getConfiguredTableName(persistentClass);
+                if (exists(databaseMetaData, tableName)) {
+                    saveTasks(connection, persistentClass, tableName);
+                }
+                return String.format("<changeSet author=\"xwikiorg\" id=\"%s0\">\n"
+                    + "  <preConditions onFail=\"MARK_RAN\">\n"
```

Comment:
Since you are already checking if the table exist, I guess you could remove that precondition and instead return the drop only in the previous if ?